checkfiles.py

```diff
@@ -1078,21 +1081,24 @@ def run(out, err, url, username, password, encValData, mirror, search_query, fil
         jobs = fetch_files(session, url, search_query, out, include_unexpired_upload, file_list, local_file)
         if not json_out:
             headers = '\t'.join(['Accession', 'Lab', 'Errors', 'Aliases', 'Upload URL',
-                                'Upload Expiration'])
+                                 'Upload Expiration'])
             out.write(headers + '\n')
             out.flush()
         for job in imap(functools.partial(check_file, config, session, url), jobs):
             if not dry_run:
                 patch_file(session, url, job)
-
-            if not job.get('skip'):
-                errors_string = str(job.get('errors', {'errors': None}))
-            else:
-                errors_string = str({'errors': 'status have not been changed, the file check was skipped due to the file unavailability on S3'})
+            errors_string = {'errors': None}
```

Comment:
I think the original problem came from the errors not being added in the correct place.  the catch all conditional in the code is the real problem

Add the errors to the job when they are found in the check_file function at 
```
except FileNotFoundError:
```
and
```
except OSError:
```
Following thats pattern then allows us to handle the errors just like the rest.
```
errors_string = {'errors': None}
if job['errors']:
    errors_string = job['errors']
```
Instead of having a special case that actually catches every job['skip'] that doesn't have errors and calls it an s3 issue.   

If we still want to catch the job['skip'] = true that don't have errors then add
```
else:
    errors_string = {'errors':  'Job skip true without error'}
```

Reply:
Fixed